ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixClusterManagerMetrics.java

```diff
@@ -190,15 +196,104 @@ void initializePartitionMetrics() {
-            logger.info("There are more than half of the replicas are down for partition {}, the down replicas are {}",
-                partition.toPathString(), downReplicas);
+          logger.info("There are more than half of the replicas are down for partition {}, the down replicas are {}",
+              partition.toPathString(), downReplicas);
           return 1L;
         }
       }
       return 0L;
     };
     registry.gauge(MetricRegistry.name(HelixClusterManager.class, "isMajorityReplicasDownForAnyPartition"),
         () -> isMajorityReplicasDownForAnyPartition);
+
+    Gauge<Integer> getLocalPartitionCountWithOneReplicaDown = () -> {
+      String dcName = clusterMapCallback.getDatacenterName();
+
+      for (PartitionId partition : clusterMapCallback.getPartitions()) {
+        // Get total replica count
+        int totalLocalReplicaCount = (int) partition.getReplicaIds()
+            .stream()
+            .filter(replicaId -> replicaId.getDataNodeId().getDatacenterName().equals(dcName))
+            .count();
+
+        // Get local leader/standby/bootstrap replica counts of this partition
+        int localLeaderStandByReplicaCount = 0;
+        for (List<? extends ReplicaId> replicaIds : partition.getReplicaIdsByStates(
+            EnumSet.of(ReplicaState.LEADER, ReplicaState.STANDBY), dcName).values()) {
+          localLeaderStandByReplicaCount += replicaIds.size();
+        }
+        int localBootstrapReplicaCount = partition.getReplicaIdsByState(ReplicaState.BOOTSTRAP, dcName).size();
+        int localInactiveReplicaCount = partition.getReplicaIdsByState(ReplicaState.INACTIVE, dcName).size();
+
+        // Calculate the number of replicas that are down. Bootstrap (upcoming) and Inactive (deactivating) replicas are considered as good.
+        int downReplicaCount = totalLocalReplicaCount - localLeaderStandByReplicaCount - localBootstrapReplicaCount
+            - localInactiveReplicaCount;
+
+        // Calculate the number of replicas that are not in leader or standby
+        int nonLeaderStandbyReplicaCount = totalLocalReplicaCount - localLeaderStandByReplicaCount;
+
+        // Update the maps to keep track of partition counts where 1, 2 or more replicas are down.
+        if (downReplicaCount > 0) {
+          localPartitionCountsByDownReplicas.put(downReplicaCount,
+              localPartitionCountsByDownReplicas.getOrDefault(downReplicaCount, 0) + 1);
+        }
+        if (nonLeaderStandbyReplicaCount > 0) {
+          localPartitionCountsByNonLeaderStandbyReplicas.put(nonLeaderStandbyReplicaCount,
+              localPartitionCountsByNonLeaderStandbyReplicas.getOrDefault(nonLeaderStandbyReplicaCount, 0) + 1);
+        }
+      }
+
+      return localPartitionCountsByDownReplicas.getOrDefault(1, 0);
+    };
+
+    // Metric to see number of local partitions with 1 replica down
+    registry.gauge(MetricRegistry.name(HelixClusterManager.class, "localPartitionCountWithOneReplicaDown"),
```

Comment:
we can probably remove "local" from the metric names (and variable names), just putting some comments here and in the ingraph, it should be fine. And do we have any metric showing number of partitions?

Reply:
Modified the metric and variable names. Yeah, we have below metric showing number of partitions..

```
    Gauge<Long> partitionCount = clusterMapCallback::getPartitionCount;
    registry.gauge(MetricRegistry.name(HelixClusterManager.class, "partitionCount"), () -> partitionCount);
```